component/org.wso2.carbon.identity.oauth2.grant.token.exchange/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/utils/TokenExchangeUtils.java

```diff
@@ -347,24 +349,41 @@ public static void setAuthorizedUser(OAuthTokenReqMessageContext tokenReqMsgCtx,
         AuthenticatedUser authenticatedUser = null;
 
         FederatedAssociationConfig federatedAssociationConfig = identityProvider.getFederatedAssociationConfig();
-        ClaimConfig claimConfig = identityProvider.getClaimConfig();
-
         ServiceProvider serviceProvider = getServiceProvider(tokenReqMsgCtx);
-        if (serviceProvider.getClaimConfig() != null &&
-                serviceProvider.getClaimConfig().isAlwaysSendMappedLocalSubjectId() &&
-                !Constants.LOCAL_IDP_NAME.equals(identityProvider.getIdentityProviderName())) {
-
-            if (federatedAssociationConfig.isEnabled()) {
-                String subjectIdentifier = resolveSubjectIdentifier(claimsSet,
-                        federatedAssociationConfig.getLookupAttributes());
-                User localUser = getLocalUser(tokenReqMsgCtx, subjectIdentifier);
-                if (localUser != null) {
-                    if (!isUserAssociated(localUser, identityProvider, subjectIdentifier)) {
-                        createAssociation(localUser, identityProvider, subjectIdentifier);
-                    }
-                    authenticatedUser = new AuthenticatedUser(localUser);
+        ClaimConfig serviceProviderClaimConfig = serviceProvider.getClaimConfig();
+        if (!Constants.LOCAL_IDP_NAME.equals(identityProvider.getIdentityProviderName())) {
+            SendLocalUser sendLocalUser = getSendLocalUser(serviceProviderClaimConfig);
```

Comment:
as mentioned in an above comment, this is not much readable. Rename the method as well (in addition to enum commented above)

Reply:
addressed with [9faa855](https://github.com/wso2-extensions/identity-oauth2-grant-token-exchange/pull/20/commits/9faa855222dd5ba4a3349ced5d28388b6c33a8ff)